File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.js
const fs = require("fs");

function solve() {
  const data = fs
    .readFileSync(process.platform === "linux" ? 0 : "input.txt", "utf-8")
    .trimEnd();
  const chunks_by_minus = data.split("-");
  const sums = chunks_by_minus.map((chunk) => {
    const chunks_by_plus = chunk.split("+");
    return chunks_by_plus.reduce((sum, element) => sum + parseInt(element), 0);
  });
  return sums.reduce((result, element) => result - element);
}

console.log(solve());

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
def sum(numbers):
    result = 0
    for number in numbers:
        result += int(number)
    return result

chunks = list(map(lambda x: sum(x.split("+")),input().split("-")))

result = chunks[0]

for i in range(1, len(chunks)):
    result -= chunks[i]

print(result)
